core/src/main/scala/cats/collections/Heap.scala

```diff
@@ -194,28 +195,28 @@ sealed abstract class Heap[A] {
   /**
    * Alias for add
    */
-  def +(x: A)(implicit order: Order[A]): Heap[A] = add(x)
+  def +[AA >: A](x: AA)(implicit order: Order[AA]): Heap[AA] = add(x)
 
   /**
    * Alias for addAll
    */
-  def ++(as: Iterable[A])(implicit order: Order[A]): Heap[A] = addAll(as)
+  def ++[AA >: A](as: Iterable[AA])(implicit order: Order[AA]): Heap[AA] = addAll(as)
 
   /**
    * Alias for remove
    */
-  def --(implicit order: Order[A]): Heap[A] = remove
+  def --[AA >: A](implicit order: Order[AA]): Heap[AA] = remove[AA]
 
   /**
    * convert to a PairingHeap which can do fast merges,
    * this is an O(N) operation
    */
-  def toPairingHeap: PairingHeap[A] =
+  def toPairingHeap[AA >: A]: PairingHeap[AA] =
```

Comment:
I don’t see why this needs to be AA if PairingHeap is covariant.

Reply:
Hmm yeah I'm not sure why I did this. I remember at one point encountering a compiler error that I was surprised by; maybe it was related to this. I'll take a look.